opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -170,275 +216,6 @@ class SpanKind(enum.Enum):
     CONSUMER = 4
 
 
-class Span(abc.ABC):
-    """A span represents a single operation within a trace."""
-
-    @abc.abstractmethod
-    def end(self, end_time: typing.Optional[int] = None) -> None:
-        """Sets the current time as the span's end time.
-
-        The span's end time is the wall time at which the operation finished.
-
-        Only the first call to `end` should modify the span, and
-        implementations are free to ignore or raise on further calls.
-        """
-
-    @abc.abstractmethod
-    def get_context(self) -> "SpanContext":
-        """Gets the span's SpanContext.
-
-        Get an immutable, serializable identifier for this span that can be
-        used to create new child spans.
-
-        Returns:
-            A :class:`.SpanContext` with a copy of this span's immutable state.
-        """
-
-    @abc.abstractmethod
-    def set_attribute(self, key: str, value: types.AttributeValue) -> None:
-        """Sets an Attribute.
-
-        Sets a single Attribute with the key and value passed as arguments.
-        """
-
-    @abc.abstractmethod
-    def add_event(
```

Comment:
Unrelated, should `Event` be in the API package? It seems like it is defined in the SDK.

Reply:
As it stands today, an event object is semi-described and alluded to in the API:

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#add-events

But to your point it is not explicitly needed in the API. We could come back here, but here's the issue from Carlos asking for it: https://github.com/open-telemetry/opentelemetry-python/issues/76

Feel free to re-raise the issue, but I'll leave this in for now.